### ISSUE TITLE
Unexpected token with type annotations

### ISSUE THREAD

**Issue by marudor** [babel/babel-eslint#201]
The following line will trigger the parsing error.
```javascript
addItem = async (parentId: string, type: string, position: ?number, title: string = '', children: ?Array) => {}
```

The problem seems to be the default value for title. If I remove the default value it works just fine.

**Comment by hzoo** [comment]
Does flow allow `title = '' : string`? Otherwise if it is a parsing issue it's probably an issue with babel then.

**Comment by marudor** [comment]
Looks like a Babel bug indeed.
I've tested the following:
The Problem is the arrow function.
Using normal function syntax:
```javascript
async function asyncTest(parentId: string, type: string, position: ?number, title: string = '', children: ?Array) {}
```
It works fine.
A non string default value also works for the arrow function. ```title: number = 42```

**Comment by hzoo** [comment]
Ok cool, and this might be already reported in babel so you should search for something like it before reporting a new issue

**Comment by hzoo** [comment]
Ok similar report was https://github.com/babel/babel-eslint/issues/190

**Comment by mmahalwy** [comment]
Why is this closed? Still a bug...

**Comment by hzoo** [comment]
It's a duplicate so it's closed - so check #190. It looks like it's was babel parser issue which might of been fixed in babel 6
